File: gateway/src/main/java/com/example/gateway/dto/UserDto.java
package com.example.gateway.dto;

import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class UserDto {
    private final Long id;

    @NotBlank(groups = {OnCreate.class}, message = "Имя пользователя не может быть пустым.")
    private final String name;

    @Email(groups = {OnCreate.class, OnUpdate.class}, message = "Некорректный электронный адрес пользователя.")
    @NotNull(groups = {OnCreate.class}, message = "Электронный адрес пользователя не может быть пустым.")
    private final String email;
}

File: api/src/main/kotlin/com/example/api/model/UserDto.kt
package com.example.api.model

import com.example.api.group.Create
import com.example.api.group.Update
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

@Schema(description = "Информация о пользователе")
data class UserDto(

    @get:Schema(description = "Идентификатор пользователя")
    val id: Long,

    @get:Schema(description = "Имя пользователя")
    @get:NotBlank(groups = [Create::class])
    val name: String?,

    @get:Schema(description = "Электронная почта пользователя")
    @get:Email(groups = [Create::class, Update::class])
    @get:NotNull(groups = [Create::class])
    val email: String?
)
